ord_schema/proto/reaction.proto

```diff
@@ -861,26 +852,104 @@ message ReactionProduct {
+    // to confirm the identity of the reported product.
+    IDENTITY = 2;
+    // Yields should only be reported as quantitative yields, e.g., measured
+    // using an isolated weight analysis, quantitative NMR with an internal
+    // standard, LC peak area with an internal standard and calibration curve
+    // obtained using an authentic standard. 
+    YIELD = 3;
+    // Selectivites include enantiomeric excess/ratios, diastereomic ratios,
+    // etc. The detailed type should be specified in selectivity_type.
+    SELECTIVITY = 4;
+    // The purity of the product mixture, normalized to all other species
+    // observed in the analysis.
+    PURITY = 5;
+    // Integrated peak area, e.g., as in an LC chromatogram.
+    AREA = 6;
+    // Retention time, e.g., as in an LC or GC chromatogram.
+    RETENTION_TIME = 7;
+    // Raw counts, e.g., as in an EIC or TIC from a MS analysis. If an EIC,
+    // the m/z ratio(s) should be defined in the details.
+    COUNTS = 8;
+    // Intensity, e.g., of a TLC spot, ELSD, or GC FID.
+    INTENSITY = 9;
+    // Direct m/z measurement, e.g., from HRMS. The value field should be used
+    // to report the m/z, not the "mass" field.
+    MZ = 10;
+    // Temperature, e.g., from a melting point experiment.
+    TEMPERATURE = 11;
+    // Angle, e.g., from a circular dichroism experiment. Units of degrees
+    // centimer squared per decimol are assumed.
+    ANGLE = 12;
   }
-  SelectivityType type = 1;
-  string details = 2;
-  optional float value = 3;
-  // TODO(kearnes): What does precision mean in this context?
-  optional float precision = 4;
+  MeasurementType type = 2;
+  // Additional information about the measurement and, in particular, how it
+  // was calculated if that is ambiguous given the analysis type.
+  string details = 3;
+  // Whether an internal standard was used for this measurement to normalize
+  // it, e.g., for quantification of yield or relative peak area.
+  optional bool uses_internal_standard = 4;
+  // Whether an authentic sample of a material was used for callibrating
+  // an analytical technique, e.g., for product identification by LCMS.
+  // "Authentic" refers to a sample of a compound whose structure has been
+  // proven by the accepted means (1H NMR, 13C NMR, IR, HRMS), and is now
+  // being used as a comparison for other analytical methods.
+  optional bool uses_authentic_standard = 5;
+  // Details about the authentic standard used, optionally including
+  // information about its precise salt form and source/vendor.
+  Compound authentic_standard = 6;
+  // The numeric value for this measurement. For example, the yield expressed
+  // as a percentage (i.e., 50% yield corresponds to 50 and not 0.5). If the
+  // value is associated with units, those unit-specific fields should be used
+  // instead.
+  optional float value = 7;
+  optional float precision = 8;
+  // The value for this measurement metric when it cannot be represented by a
+  // single number.
```

Comment:
"e.g. selectivity as '2:1'"?

Reply:
Will add

Reply:
Do we want to give guidance on how this should be used? E.g., where "2:1" is (desired / all others) _or_ if there are more than two then use, e.g., "2:1:1", which would be "1:1" desired / all others (in the `SelectivityType` message it's written as "desired / undesired")? Also, how do you envision this will be handled downstream if some people enter this string, some enter "67:33" and others use `float value` 2? Some might also say the _selectivity value_ of a reaction giving a 2:1 product mixture is really 67%, which would be handled by `float value` as 67 or 0.67... Is it worth creating a structure to standardize all this?

Reply:
I think we should be clear about expected usage. In the case you brought up, the ambiguity between 67, 0.67, and 2 wouldn't be addressed by the fields themselves since all three are floats, but by the comments. 

If someone were to be tempted to enter "2:1:1", we would need to know what the other species were (unless it was somehow obvious/unambiguous, which I think could be true if there are only two species 2:1 but not for more than two). In that case, I'm not actually sure that we care about "2:1:1" -- we would rather have the area/counts of each product as a separate measurement, which could be combined to get the relative yields. "2:1:1" is after an additional stage of processing. If that information isn't available (e.g., entering an already published reaction), then I think associating "2:1:1" with the first/primary product and having the other numbers correspond to other listed products, in order, might make sense.

Reply:
I agree, though this does raise the question of whether `SELECTIVITY` is necessary at all. If we plan to derive it downstream from the _yields_ of multiple species anyway (which I think is sensible), we then expect that each species is defined as a product and quantified. Selectivity would then be inherently extractable. 

That said, I can see the value of a quick call to the selectivity metric, so long as usage is consistent. It is also a clear marker of, e.g., enantio- or diastereoselective reactions, if looking to learn there.

Reply:
Elaborated in a comment.